tests/queries/0_stateless/03287_dynamic_and_json_squashing_fix.sql

```diff
@@ -0,0 +1,24 @@
+set enable_json_type=1;
+set enable_dynamic_type=1;
+
+drop table if exists src;
+drop table if exists dst;
+
+create table src (d Dynamic) engine=Memory;
+create table dst (d Dynamic) engine=MergeTree order by tuple();
+insert into src select materialize(42)::Int64;
+insert into src select 'Hello';
+insert into dst select * from remote('127.0.0.2', currentDatabase(), src);
```

Comment:
Quick question. Why `127.0.0.2`? Is it a typo?

Reply:
Not necessarily, see https://github.com/ClickHouse/ClickHouse/blob/5cfaf4e4d6a2986c8039fe61acdb038a53d4607c/src/Common/isLocalAddress.cpp#L84-L92

Reply:
I just wanted to avoid optimization `prefer_localhost_replica` that won't use the native protocol that I need to test